scripts/build/build_darwin_framework.py

```diff
@@ -89,6 +86,16 @@ def build_darwin_framework(args):
                         default="Matter",
                         help="Name of target to build",
                         required=True)
+    parser.add_argument("--target_sdk",
+                        default="macosx",
+                        help="Set the target sdk",
```

Comment:
Are these fixed values? It would be nice if help describes possible values.

Asking because a conversion to build.py would result in variants such as `darwin-framework-arm-macosx` or `darwin-framework-x64-ios` or similar. It is useful to know when something is free form (like some SDK path which cannot be a variant) vs just a subset of options.

Same question for below.

Reply:
> Are these fixed values? It would be nice if help describes possible values.
> 
> Asking because a conversion to build.py would result in variants such as `darwin-framework-arm-macosx` or `darwin-framework-x64-ios` or similar. It is useful to know when something is free form (like some SDK path which cannot be a variant) vs just a subset of options.

Ideally I was hoping to play with `--isysroot` as one may want to specify a directory. But it will require some deeper changes that I need to be careful with. In general our usage of the gn specifics bits for building Xcode applications can be improved.